### PR TITLE
MMT-2067 Updating schema to include a column for submitter

### PR DESCRIPTION
**Pull request by ryanmiller-1**
And updating sorting on the various index/manage proposal pages to enable sorting by submitter.
Migration to add the user_id of the person who created the record if there is no submitter_id.  This decision was made because it simplified the migration and did not add urs_uid's to our repo.

### CODE REVIEW

File: app/concerns/proposal_index.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      @urs_user_hash[user['uid']] = "#{user['first_name']} #{user['last_name']}"
+    end
+  end
+
+  def sort_by_submitter(proposals)
+    @query = {}
+    @query['sort_key'] = params['sort_key'] unless params['sort_key'].blank?
+
+    sorted_proposals = proposals.all.sort do |a, b|
+                         # Making these arrays allows empty submitter_ids to sort last/first for ASC/DESC
```
Comment (ryanmiller-1) [inline app/concerns/proposal_index.rb]: expand this comment

File: app/controllers/proposal/collection_draft_proposals_controller.rb [hunk 2]
```diff
@@ -1,35 +1,55 @@
...
     before_action(only: [:submit, :rescind, :progress, :approve, :reject]) { set_resource }
 
     def index
-      resources = resource_class.order(index_sort_order)
-                                .page(params[:page]).per(RESULTS_PER_PAGE)
-      instance_variable_set("@#{plural_resource_name}", resources)
+      working_proposals = resource_class.all
+      set_urs_user_hash(working_proposals)
+      if params['sort_key']&.include?('submitter_id')
+        resources = sort_by_submitter(working_proposals)
```
Comment (ryanmiller-1) [inline app/controllers/proposal/collection_draft_proposals_controller.rb]: investigate helper function

File: app/views/proposal/collection_draft_proposals/index.html.erb [hunk 3]
```diff
@@ -43,7 +43,7 @@
               <%= "#{proposal.request_type.titleize}" %>
             </td>
             <td>
-              <%= proposal['status_history'].fetch('submitted', proposal.in_work? ? { 'username' => 'Pending' } : { 'username' => 'Not Provided' })['username'].titleize %>
+              <%= @urs_user_hash[proposal.submitter_id] || 'Pending' %>
```
Comment (ryanmiller-1) [inline app/views/proposal/collection_draft_proposals/index.html.erb]: Helper method

File: app/views/manage_proposal/show.html.erb [hunk 4]
```diff
@@ -59,7 +59,7 @@
               <%= proposal['request_type'].titleize %>
             </td>
             <td>
-              <%= proposal['status_history'].fetch('submitted', { 'username' => 'Not Provided' })['username'].titleize %>
+              <%= @urs_user_hash[proposal['submitter_id']] || 'Pending' %>
```
Comment (ryanmiller-1) [inline app/views/manage_proposal/show.html.erb]: See index erb

File: app/concerns/proposal_index.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  def set_urs_user_hash(proposals)
+    submitters = proposals.map { |proposal| proposal['submitter_id'] }.uniq
+    urs_users = retrieve_urs_users(submitters)
+    @urs_user_hash = {}
+    urs_users.each do |user|
+      @urs_user_hash[user['uid']] = "#{user['first_name']} #{user['last_name']}"
+    end
+  end
+
+  def sort_by_submitter(proposals)
```
Comment (charleshuang80) [inline app/concerns/proposal_index.rb]: I know it must be set before this method is called, but I think we should pass in `@urs_user_hash` as a parameter to this function.

File: app/helpers/proposals_helper.rb [hunk 6]
```diff
@@ -127,4 +127,8 @@ def get_available_actions_text
       'No actions are possible.'
     end
   end
+
+  def display_submitter_name(submitter_id)
+    @urs_user_hash[submitter_id] || 'Pending'
```
Comment (charleshuang80) [inline app/helpers/proposals_helper.rb]: same comment, I think we should pass in `@urs_user_hash` as a parameter

**Pull request merged**